src/domains/shielder/utils/useShielderFees.tsx

```diff
@@ -1,32 +1,161 @@
 import { erc20Token, nativeToken } from '@cardinal-cryptography/shielder-sdk';
 import { skipToken, useQuery } from '@tanstack/react-query';
-import { Address } from 'viem';
+import { useMemo } from 'react';
+import { z } from 'zod';
 
+import { useWallet } from 'src/domains/chains/components/WalletProvider.tsx';
 import { Token } from 'src/domains/chains/types/misc';
 import useChain from 'src/domains/chains/utils/useChain';
 import getQueryKey from 'src/domains/misc/utils/getQueryKey';
+import useEstimateAllowanceFee from 'src/domains/shielder/utils/useEstimateAllowanceFee';
 import useShielderClient from 'src/domains/shielder/utils/useShielderClient';
 
+export type FeeItem = {
+  type: 'network' | 'relayer' | 'allowance',
+  amount: bigint,
+  token: string,
+};
+
+export type FeeStructure = FeeItem[];
+
+const FeeEstimationSchema = z.object({
```

Comment:
Not a component, nor a class, so
```suggestion
const feeEstimationSchema = z.object({
```